kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java

```diff
@@ -639,6 +645,33 @@ public String toString() {
     }
   }
 
+  private static boolean checkBucketExists(String bucketName) {
+    try {
+      final AmazonS3 bucketCheckHelper = AmazonS3ClientBuilder.defaultClient();
+      return bucketCheckHelper.doesBucketExist(bucketName);
+    } catch (SdkClientException e) {
+      return false;
```

Comment:
I am not sure about how this exception is being caught, is this a network error? maybe we should bubble up the actual exception?

Reply:
How does the `AmazonS3ClientBuilder` know how to get the credentials? Remember, the `` property might specify a different credential provider class(es), which the `AmazonS3ClientBuilder` may not know about. Or, the `AmazonS3ClientBuilder` might use multiple credential providers but in a different order -- in that case, a user might have locally set them one way (e.g., config file) and still had other credentials set earlier but a different way (e.g., env vars). If the default credentials provider used by the `AmazonS3ClientBuilder` doesn't check them in the same order as the user-defined credentials provider class, then the wrong credentials might be used.

The bottom line is that the Validator on the `s3.bucket.name` only knows the value of that property and simply does not know the value for any other property. And because `s3.credentials.provider.class` is a different property, the `s3.bucket.name` validator can't actually assume the credentials are accessed in any particular way, since that might differ from what the `s3.credentials.provider.class` class says.